java/core/src/main/java/com/redhat/rhn/frontend/action/systems/sdc/SystemDetailsEditAction.java

```diff
@@ -220,24 +222,29 @@ else if (daForm.get(AUTO_UPDATE) == null) {
 
         if (scheduleId != null && scheduleId != 0) {
             // Assign schedule
-            MaintenanceSchedule schedule = maintenanceManager.lookupScheduleByUserAndId(user, scheduleId).get();
-            boolean cancelAffected = Boolean.TRUE.equals(daForm.get(MAINTENANCE_CANCEL_AFFECTED));
             try {
+                //if optional.get fails, it's caught in the catch part
+                MaintenanceSchedule schedule =
+                        maintenanceManager.lookupScheduleByUserAndId(user, scheduleId).get(); //NOSONAR false positive
```

Comment:
Be explicit about wanting to throw the exception instead of disabling SonarCloud.
```suggestion
                        maintenanceManager.lookupScheduleByUserAndId(user, scheduleId).orElseThrow();
```